### ISSUE TITLE
Android 11 compatibility logcat error

### ISSUE THREAD

**Issue by perracodex** [google/ExoPlayer#8363]
- ExoPlayer 2.12.2
- Android 11 (R)

In Android 11 the next error is shown in the logcat:

```
Tried to access visual service WindowManager from a non-visual Context:app.application.App@35e268c Visual services, such as WindowManager, WallpaperService or LayoutInflater should be accessed from Activity or other visual Context. Use an Activity or a Context created with Context#createWindowContext(int, Bundle), which are adjusted to the configuration and visual bounds of an area on screen.
```

The problem is originated from the fact that in the ExoPlayer **MediaCodectVideoRenderer** class constructor, the context is resolved as the application one:

`this.context = context.getApplicationContext();`

Full logcat stacktrace:

```
E/ContextImpl: Tried to access visual service WindowManager from a non-visual Context:app.application.App@35e268c Visual services, such as WindowManager, WallpaperService or LayoutInflater should be accessed from Activity or other visual Context. Use an Activity or a Context created with Context#createWindowContext(int, Bundle), which are adjusted to the configuration and visual bounds of an area on screen.
    java.lang.IllegalAccessException: Tried to access visual service WindowManager from a non-visual Context:app.application.App@35e268c
        at android.app.ContextImpl.getSystemService(ContextImpl.java:1914)
        at android.content.ContextWrapper.getSystemService(ContextWrapper.java:803)
        at com.google.android.exoplayer2.video.VideoFrameReleaseTimeHelper.<init>(VideoFrameReleaseTimeHelper.java:77)
        at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.<init>(MediaCodecVideoRenderer.java:255)
        at com.google.android.exoplayer2.DefaultRenderersFactory.buildVideoRenderers(DefaultRenderersFactory.java:366)
        at com.google.android.exoplayer2.DefaultRenderersFactory.createRenderers(DefaultRenderersFactory.java:310)
        at com.google.android.exoplayer2.SimpleExoPlayer.<init>(SimpleExoPlayer.java:610)
        at com.google.android.exoplayer2.SimpleExoPlayer$Builder.build(SimpleExoPlayer.java:513)
        at app.ui.widget.video_player.SessionPlayback.<init>(SessionPlayback.java:97)
        at app.ui.widget.video_player.VideoPlayerView.setMedia(VideoPlayerView.java:195)
```

**Comment by tonihei** [comment]
Could you provide some context abut how and where you use the player? It sounds as if you are using it somewhere where the application isn't an "Activity or other visual Context" (maybe some kind of Service?), but it still tries to render video. It would be good to understand how we can reproduce the issue if needed. Otherwise it's hard to know whether this needs fixing and if yes, how it should be fixed.

**Comment by perracodex** [comment]
Hello, thanks for the quick response.

My implementation is executed 100% only in the UI thread, but notice that **this error is shown in the logcat even when running in the UI thread "whenever using an application context with the WindowsManager"**. 
The error is new and was introduced in Android 11, hence the reason why doesn't appear with older android versions.

Debugging ExoPlayer I have found where the problem happens exactly.  Is specifically in the **VideoFrameReleaseTimeHelper** constructor. See screenshot:

![image](https://user-images.githubusercontent.com/7354233/102647164-99a28780-4165-11eb-8bb5-52048640203e.png)

Therefore this explains the meaning of the error message:

`Tried to access visual service WindowManager from a non-visual Context`

In addition, I've also noticed that in the **MediaCodecVideoRenderer** constructor the referenced instance context is set to the application one. I couldn't find if this has any impact for this specific case, but may be good to keep in mind that then this context shouldn't be used neither to access the WindowsManager if needed. See also a screenshot:

![image](https://user-images.githubusercontent.com/7354233/102647398-f8680100-4165-11eb-8e5c-9e5e499cfeb8.png)
